pandas/tests/io/test_stata.py

```diff
@@ -499,7 +499,7 @@ def test_timestamp_and_label(self, version):
 
             with StataReader(path) as reader:
                 assert reader.time_stamp == '29 Feb 2000 14:21'
-                assert reader.data_label == data_label
+                assert reader._data_label == data_label
```

Comment:
Revert this

Reply:
And finally, use (line 1734)
```
@property
def data_label():
```

Reply:
@bashtage @WillAyd Does the property need to return `self._data_label` instead of `self.data_label` here ? Because, I get a recursion error if I don't  (see below) .............


self = <pandas.tests.io.test_stata.TestStata object at 0x1c1d86d128>, version = 117

    @pytest.mark.parametrize('version', [114, 117])
    def test_timestamp_and_label(self, version):
        original = DataFrame([(1,)], columns=['variable'])
        time_stamp = datetime(2000, 2, 29, 14, 21)
        data_label = 'This is a data file.'
        with tm.ensure_clean() as path:
            original.to_stata(path, time_stamp=time_stamp,
                              data_label=data_label,
                              version=version)
    
            with StataReader(path) as reader:
                assert reader.time_stamp == '29 Feb 2000 14:21'
>               assert reader.data_label == data_label

pandas/tests/io/test_stata.py:502: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
pandas/io/stata.py:1739: in data_label
    return self.data_label
pandas/io/stata.py:1739: in data_label
    return self.data_label
E   RecursionError: maximum recursion depth exceeded
!!! Recursion detected (same locals & position)

Reply:
Yes, the assignments need to be to `self._data_label = self._get...`, and then the property should 

```
@property
def data_label(self):
    """"Docstring"""
    return self._data_label
```

Reply:
Cheers, just pushed latest updates